### ISSUE TITLE
Detect usage of `Symfony\Component\Yaml\Yaml::parse` and suggest `Drupal\Component\Serialization\Yaml::decode`

### ISSUE THREAD

**Issue by Kingdutch** [mglaman/phpstan-drupal#642]
# Feature request
When developers work with Yaml the autocomplete from IDEs often suggests `Yaml::parse` which will give them the Symfony component and they may be unaware of the existence of Drupal's Yaml serialization component which is strictly better.

Developers should prefer calling `Drupal\Component\Serialization\Yaml::decode` which can delegate to Symfony, but will prefer the PHP Yaml extension if it's available which provides a free performance benefit.

**Comment by mglaman** [comment]
Actually, it should recommend https://git.drupalcode.org/project/drupal/-/blob/11.x/core/lib/Drupal/Core/Serialization/Yaml.php?ref_type=heads

I know. We have two. But this respects the ability to override it

**Comment by mglaman** [comment]
@Kingdutch actually, I think we should open a Drupal core issue to kill this class. `Drupal\Component\Serialization\Yaml` should have a `setSerializer` method. `DrupalKernel` as part of `bootEnvironment` process could call the method to set the serializer.


```php
Drupal\Component\Serialization\Yaml::setSerializer(Settings::get('yaml_parser_class'))
```

What do you think?

**Comment by mglaman** [comment]
I went an opened one https://www.drupal.org/project/drupal/issues/3401665

**Comment by Kingdutch** [comment]
Yeah that sounds like a good Drupal core issue :) I still agree with the title change to get people to switch to the most flexible class first, then they'll eventually automatically get the "deprecation" warning to move to the component when that is more flexible.

We'll need to include some logic in the rule that suggests `Drupal\Core\Serialization\Yaml`, to check whether that class is not deprecated and suggest `Drupal\Component\Serialization\Yaml` otherwise (or we risk people getting stuck between a rock and a hard place when the deprecation lands in Drupal).

**Comment by mglaman** [comment]
🤖 

Revisited the linked Drupal.org issues to check current status:

- [#3205480](https://www.drupal.org/project/drupal/issues/3205480) was **closed (fixed)** in Drupal 10.3/11.x — PECL YAML support was dropped and `Drupal\Core\Serialization\Yaml` was converted to a plain `class_alias()` pointing to `Drupal\Component\Serialization\Yaml`.
- [#3401665](https://www.drupal.org/project/drupal/issues/3401665) (the issue I opened) was closed as a duplicate of the above. The full deprecation of `Drupal\Core\Serialization\Yaml` was **not** implemented — it remains a simple alias.

This actually simplifies the rule. The concern from @Kingdutch about needing branching logic ("check if `Core\Serialization\Yaml` is deprecated, suggest `Component` otherwise") is no longer needed. The two classes are now functionally identical.

The rule is still worth implementing: detect direct usage of `Symfony\Component\Yaml\Yaml::parse` and suggest `Drupal\Component\Serialization\Yaml::decode` instead. Using Symfony's class directly bypasses Drupal's `yaml_parser_class` setting, so the nudge is still valid.
